index.js

```diff
@@ -0,0 +1,94 @@
+const BASE_URL = 'https://trektravel.herokuapp.com';
+$(document).ready(() => {
+
+  $('#load').click(function(response) {
+
+    let fullUrl = BASE_URL + '/trips'
+    $.get(fullUrl, response => {
+      console.log(response);
+
+      response.forEach((trip) => {
+        let name = trip.name,
+        id = trip.id,
+        continent = trip.continent,
+        weeks = trip.weeks;
+        let allTripInfo = `<li data-id=${id}><h3>${name}:</h3> <p> Location: ${continent} Length: ${weeks} weeks. </p></li>`;
+        $('#all-trips ul').append(allTripInfo);
+      });
+
+      $('#all-trips h3').on('click', onTripClick)
+    })
+    .fail(function(response){
+      console.log(response);
+      console.log('failure');
+      $('#message').html(`<p>Error!</p>`)
+    })
+    .always(function(){
+      console.log('always even if we have success or failure');
+    });
```

Comment:
I'm not sure how useful the `always` callback is here. I think of `always` like the `ensure` clause of a `begin/rescue` block: every once in a very long while it's incredibly useful, and otherwise I forget it's even there.